sdk/src/main/scala/io/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1003,7 +1003,7 @@ class EthService(
       // limit the range of blocks by the number of available blocks and cap at 1024
       val blocks = blockCount.intValueExact().min(requestedBlockInfo.height).min(1024)
       // geth comment: returning with no data and no error means there are no retrievable blocks
-      if (blocks < 1) {
+      if (blocks < 1 || newestBlock == "pending") {
```

Comment:
That's OK but it is better to move the check on "pending" before calling `getBlockByTag` (line 1002). This because `getBlockByTag` will create a pending block and it is an expensive operation that it is not needed anymore.

Reply:
Done, moved out of `applyOnAccountView`